### PR TITLE
Added logic for shuffle, isSet, hasValidSet

### PR DESCRIPTION
**Pull request by Silly06**

### CODE REVIEW

**Comment by nathyong** [comment]
Looks good, we can add some tests for this as well.

File: src/index.ts [hunk 1]
```diff
@@ -95,11 +105,56 @@ function gameSelectCard(card: Card): void {
...
 function isSet(cards: Set<Card>): boolean {
-    return true;
+    // mini function that checks if that an attribute is valid, i.e. all the same, or all different
+    function validAttribute(levels: Array<number>){
+        if ((levels[0] == levels[1] && levels[1] == levels[2])||(levels[0] != levels[1] && levels[0] != levels[2] && levels[1] != levels[2])){
+            return true
+        }
+        else{
+            return false
+        }
```
Comment (nathyong) [inline src/index.ts]:
```suggestion
        return ((levels[0] == levels[1] && levels[1] == levels[2])||(levels[0] != levels[1] && levels[0] != levels[2] && levels[1] != levels[2]))
```

you accidentally wrote something in the form `if (true) {return true} else {return false}`, so just return it directly.

File: src/index.ts [hunk 2]
```diff
@@ -95,11 +105,56 @@ function gameSelectCard(card: Card): void {
...
+    let shapes : Array<number> = cards[0].shape
+    let counts : Array<number> = cards[0].count
+    let shades : Array<number> = cards[0].shade
+    let colors : Array<number> = cards[0].color
+    //checks if all 4 attributes are valid, and if they are, returns true for valid set
+    if(validAttribute(shapes) && validAttribute(counts) && validAttribute(shades) && validAttribute(colors)){
+        return true;
+    }else{
+        return false
+    }
```
Comment (nathyong) [inline src/index.ts]: Same as above comment.

File: src/index.ts [hunk 3]
```diff
@@ -95,11 +105,56 @@ function gameSelectCard(card: Card): void {
...
+                }
+            }    
+        }
+    }
+    if(validSetCount > 0){
+        return true
+    }
+    else{
+        return false
+    }
```
Comment (nathyong) [inline src/index.ts]: Same as above comment.

**Review by Silly06 — COMMENTED** [review]
Updated the redundant if() return true else return false statements

**Review by nathyong — APPROVED** [review]
Looks good. Could be cleaner, but this seems correct.

I can add an example of tests in another change.

**Pull request merged**